integrationtests/chrome_test.go

```diff
@@ -2,234 +2,144 @@ package integrationtests
-			0x01, 0x03, 0x00, 0x00, 0x00, 0xb6, 0x30, 0x2a, 0x2e, 0x00, 0x00, 0x00,
-			0x03, 0x50, 0x4c, 0x54, 0x45, 0x5a, 0xc3, 0x5a, 0xad, 0x38, 0xaa, 0xdb,
-			0x00, 0x00, 0x00, 0x0b, 0x49, 0x44, 0x41, 0x54, 0x78, 0x01, 0x63, 0x18,
-			0x61, 0x00, 0x00, 0x00, 0xf0, 0x00, 0x01, 0xe2, 0xb8, 0x75, 0x22, 0x00,
-			0x00, 0x00, 0x00, 0x49, 0x45, 0x4e, 0x44, 0xae, 0x42, 0x60, 0x82,
-		})
-	})
+const (
+	nChromeRetries = 8
+)
+
+func getChromePath() string {
+	if runtime.GOOS == "darwin" {
+		return "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome"
+	}
+	return "google-chrome"
+}
 
-	http.HandleFunc("/tiles", func(w http.ResponseWriter, r *http.Request) {
-		io.WriteString(w, "<html><body>")
-		for i := 0; i < nImgs; i++ {
-			fmt.Fprintf(w, `<img src="/tile?cachebust=%d">`, i)
+func chromeTest(version protocol.VersionNumber, url string, blockUntilDone func()) {
+	// Chrome sometimes starts but doesn't send any HTTP requests for no apparent reason.
+	// Retry starting it a couple of times.
+	for i := 0; i < nChromeRetries; i++ {
+		if chromeTestImpl(version, url, blockUntilDone) {
+			return
 		}
-		io.WriteString(w, "</body></html>")
-	})
+	}
+	Fail("Chrome didn't hit the testing endpoints")
 }
 
-var _ = Describe("Chrome tests", func() {
-	// test disabled since it doesn't work with the configurable QUIC version in the server
-	PIt("does not work with mismatching versions", func() {
-		versionForUs := protocol.SupportedVersions[0]
-		versionForChrome := protocol.SupportedVersions[1]
-
-		supportedVersionsBefore := protocol.SupportedVersions
-		protocol.SupportedVersions = []protocol.VersionNumber{versionForUs}
-		wd := getWebdriverForVersion(versionForChrome)
-
-		defer func() {
-			protocol.SupportedVersions = supportedVersionsBefore
-			wd.Close()
-		}()
-
-		err := wd.Get("https://quic.clemente.io/hello")
-		Expect(err).NotTo(HaveOccurred())
-		source, err := wd.PageSource()
-		Expect(err).NotTo(HaveOccurred())
-		Expect(source).ToNot(ContainSubstring("Hello, World!\n"))
-	})
+func chromeTestImpl(version protocol.VersionNumber, url string, blockUntilDone func()) bool {
+	userDataDir, err := ioutil.TempDir("", "quic-go-test-chrome-dir")
+	Expect(err).NotTo(HaveOccurred())
+	defer os.RemoveAll(userDataDir)
+	path := getChromePath()
+	args := []string{
+		"--disable-gpu",
+		"--no-first-run=true",
+		"--no-default-browser-check=true",
+		"--user-data-dir=" + userDataDir,
+		"--enable-quic=true",
+		"--no-proxy-server=true",
+		"--origin-to-force-quic-on=quic.clemente.io:443",
+		fmt.Sprintf(`--host-resolver-rules=MAP quic.clemente.io:443 localhost:%s`, port),
+		fmt.Sprintf("--quic-version=QUIC_VERSION_%d", version),
+		url,
+	}
+	utils.Infof("Running chrome: %s '%s'", getChromePath(), strings.Join(args, "' '"))
+	command := exec.Command(path, args...)
+	session, err := gexec.Start(command, GinkgoWriter, GinkgoWriter)
+	Expect(err).NotTo(HaveOccurred())
+	defer func() {
```

Comment:
Done. There used to be a Wait() in there, but that didn't work (for some reason).